### DISCUSSION TITLE
Eating Crow as My UCTRONICS Adapters are Now Working Again

### DISCUSSION THREAD

**Discussion by Bob-AOL** [gtxaspec/wz_mini_hacks#441] [General]
Those who follow the USB/Ethernet discussions may have run across my recent frustration with not being able to bring up the UCTRONICS adapters using what in the past was a simple slam dunk initialization process that worked every time with every one of the devices I bought. 

My recent experience hasn't been nearly as smooth as my last one. Far from it. Nothing worked. Now it does. And I have no idea how or why I encountered so much difficulty or what I did to correct it. Many factors could have contributed and I've identified a few that could possibly benefit from developer review (and greater end-user attention to detail on my part), but not to the extent of my bellyaching about it during the more heightened levels of frustration along the way. 

Apologies.

**Comment by ljmerza** [comment]
i have 5 of these things and only 2 work. any ideas would be great.

**Reply by Bob-AOL** [reply]
Well then, let's see if we can get a third one going.

Which cameras and which firmware version for starters?

**Reply by Bob-AOL** [reply]
Linux or Windows?

**Comment by bjs-pdx** [comment]
I haven't tried this yet, so thought I'd ask.  I have the uctronics adapter which I tested and got working.  I have updated all my cameras to the latest Wyze firmware but want to know if I can still use the hack, just for ethernet/POE.  Not interested in RTSP or the other features at this time, which I have read won't work with the Wyze firmware.  

Any input would be appreciated.

**Reply by Bob-AOL** [reply]
Yes. I happen to be partial to the Wyze Beta 4.61.0.3 RTSP firmware over the wz_mini RTSP feature and only use the hack for the USB/ETH/POE functions. Been rock solid for me on six v3s since back when we started messing around with the adapters. (I gave up on the v2s and sent them on down the road.)

The key to the whole setup is realizing it's not quite as plug and play as implied in the README _**until**_ the adapter has been initialized as its own piece of hardware on the network with its own IP address and the camera's previous identity on the network ceases to exist. This happens the moment it is rebooted from Wi-Fi mode after enabling and saving the USB/ETH function in wz_mini.conf and _**must**_ be done live online to initiate dhcp discovery for the MAC address of the adapter to obtain an IP address. The camera then simply becomes a client behind the adapter like any other ethernet device on the network. The good news is the IP address of the adapter will be reflected in the Wyze app making it easy to find on the network. The Wi-Fi MAC will remain listed even though it's no longer relevant unless Wi-Wi mode is re-activated.

I'm working on a tutorial (of sorts) on the subject which is primarily aimed toward Windows users, but could apply to anyone wanting to explore the hack's POE/ETH feature. The thrust is to try covering everything that screws folks up or scares em enough to not try the hack at all. Not ready for prime time and I'm not sure what the protocol is here for tossing something like that out to the public. Be more than happy to send it your way in its current state if you'd care to provide an email address.

**Reply by bjs-pdx** [reply]
Trying to find a way to send a DM with my email.  Rather not post it here.  GitHub doesn't seem to allow direct messaging.  Do you have a Wyze forum account?

**Reply by Bob-AOL** [reply]
Yes. PogeSmit

**Reply by Bob-AOL** [reply]
> I have updated all my cameras to the latest Wyze firmware...

Just a note that 4.36.9.139 is the latest firmware known to allow all wz_mini features to behave as expected. The newest version is said to create issues for many of the features and requires a rollback to .139, though the USB/ETH/POE function is evidently not affected if using the later firmware -- which has also been said to have been recalled in favor of rolling back to 4.36.9.139 anyway.  

Early discussions about it are over in #355.
